include/Megamix/Types.hpp

```diff
@@ -235,13 +235,288 @@ namespace Megamix {
         Color8 unk3;
 
         MuseumRowColor(Color8 background, Color8 edgeFade) {
-            this->unk1 = { 0x00, 0x00, 0x00, 0x00 };
-            this->unk3 = { 0xFF, 0xFF, 0xFF, 0x6E };
+            this->unk1 = {0x00, 0x00, 0x00, 0x00};
+            this->unk3 = {0xFF, 0xFF, 0xFF, 0x6E};
 
             this->background = background;
             this->edgeFade = edgeFade;
         }
     };
-}
 
-#endif
\ No newline at end of file
+    struct CList {
+        int *_vtable;
+        struct CList *next;
+        struct CList *prev;
+    };
+
+    struct TickFlowStack {
+        u32 *tick;
+        u32 tickSize;
+    };
+
+    struct CTickflow {
+        CList at0;
+        u32 *tickflow;
+        u32 category;
+        int tickFlowSize;
+        float rest;
+        u8 unk1;
+        u32 condvar;
+        u32 condvarStack[10];
+        u8 pad1[0x17];
+        u8 condvarStackCounter;
+        struct TickFlowStack tickFlowStack[8];
+        u8 pad2[0x3F];
+        u8 tickFlowStackCounter;
+        u8 buttonPromptIsReleased;
+        u8 buttonPromptIsPressed;
+        u32 buttonPromptButton;
+        int buttonPromptPressSfx;
+        int buttonPromptReleaseSfx;
+        u8 pad3[7];
+        u32 countdownValue;
+        u32 countdownCounter;
+    };
+
+    //-----------------
+    //Save File
+    //-----------------
+
+    enum class EGameRank : u8 {
+        Locked = 0,
+        Unplayed = 1,
+        Unfinished = 2,
+        Not_Good = 3,
+        Ok = 4,
+        High = 5,
+        Perfect = 6
+    };
+
+    enum class EPlayStyle : u8 { /* simple tap / button */
+    Button = 0,
+    Tap = 1
+    };
+
+    struct CSaveFileData {
+    u8 pad1[0x10];
+    u32 playTime;
+    u8 playerData; /* Created by retype action */
```

Comment:
also please remove all these "create by retype action" comments since they're Ghidra-generated and could be confusing (there's some more down below but you can just Ctrl-F them)